05-Jason-Luo/wk07/1-mon/GoogleBooks/js/googlebooks.js

```diff
@@ -0,0 +1,43 @@
+const bookSearch = function (input) {
+  var search = document.getElementById("bookList").value;
+
+  const xhr = new XMLHttpRequest();
+
+  xhr.onreadystatechange = function () {
+    if (xhr.readyState !== 4) {
+      return;
+    }
+
+    const data = JSON.parse(xhr.responseText);
+    const title = data.items[0].volumeInfo.title;
+    const author = data.items[0].volumeInfo.authors[0];
+    const picture = data.items[0].volumeInfo.imageLinks.thumbnail;
+
+    // create a DOM node
+    const b1 = document.createElement("p");
+    const b2 = document.createElement("p");
+    const img = document.createElement("img");
+
+    //put the text in the DOM node
+    b1.innerHTML = `Title: ${title}`;
+    b2.innerHTML = `Author: ${author}`;
+    img.setAttribute("src", picture);
+
+    //attach the DOM node to the page
+
+    document.getElementById("display").appendChild(b1);
+    document.getElementById("display").appendChild(b2);
+    document.getElementById("display").appendChild(img);
```

Comment:
I'd fetch this element once, store it in a variable then re-use that variable instead of making repeated queries:
```suggestion
const display = document.getElementById("display");
[b1, b2, img].forEach(child => {
    display.appendChild(child);
});
```